### PR TITLE
add $ method in SpatialExperiment, closes #107

### LINKED ISSUE #107: colData access
Hi, I am wondering why, with respect to the SingleCellExperiment, the $ operator does not list the available colData entries any more but rather the rownames? I see that one can still call colData entries, which in itself is a bit mysterious to me how you can call row- and colData entries through the $ operator, but not showing the options is tedious when working with many colData entries.
I feel that is a feature that particularly hindering me from using the SPE object over the SCE object when working with spatial data.
Thanks for any thoughts on this.

### PR DESCRIPTION
**Pull request by LiNk-NY**
Hi Dario, @drighelli 
This will allow you to see the `colData` names using `$` in RStudio. 
Best regards,
Marcel

### CODE REVIEW

**Comment by LiNk-NY** [comment]
Note. You may want to add a replacement method as well `$<-`.

**Comment by drighelli** [comment]
Wow thanks @LiNk-NY this is amazing! :)

File: R/SpatialExperiment-methods.R [hunk 1]
```diff
@@ -267,3 +267,15 @@ setMethod("scaleFactors",
...
+
+#' @export
+#' @importFrom utils .DollarNames
+.DollarNames.SpatialExperiment <- function(x, pattern = "")
+    grep(pattern, names(colData(x)), value = TRUE)
+
+#' @rdname SpatialExperiment-methods
+#' @aliases $,SpatialExperiment-method
+#' @exportMethod $
+setMethod("$", "SpatialExperiment", function(x, name) {
```
Comment (drighelli) [inline R/SpatialExperiment-methods.R:279]: please provide documentation for the arguments of the exported methods

**Review by drighelli — COMMENTED** [review]
@LiNk-NY could you please provide documentation for the setter method arguments? 

It seems that the missing documentation produces an error during the checks.

Thanks

**Review by drighelli — APPROVED** [review]

**Comment by drighelli** [comment]
Thanks @LiNk-NY, nice work! 
let's wait for the others, they can have additional comments :)

**Review by lmweber — APPROVED** [review]
Yes, this looks great, thank you very much! This is a nice simple solution and seems to work fine when I tested it out.

I'll try to add a version bump and NEWS in this PR by checking it out locally and pushing into the PR, and then we can merge.

**Comment by lmweber** [comment]
I added an extra period in the docs for consistency, and the news item and version bump. If this looks ok I think we can merge. Thank you!

**Comment by lmweber** [comment]
(The failed build for Windows from GitHub Actions seems to be some unrelated issue due to GitHub Actions. Maybe we can think about dropping the Windows and Mac builds from GitHub Actions later, if these are causing unnecessary problems.)

**Comment by drighelli** [comment]
Thanks @lmweber,

I also think this windows issue is unrelated to the PR, but I didn't remove the different platforms from the GHA, maybe this will fix by themself with some updates.

**Pull request merged**